File: ruby/logbrew-ruby/examples/real_user_smoke.rb
# frozen_string_literal: true

require "json"

begin
  require "logbrew"
rescue LoadError
  require_relative "../lib/logbrew"
end
require_relative "readme_example"

client = LogBrew::Client.create(
  api_key: "LOGBREW_API_KEY",
  sdk_name: "logbrew-ruby",
  sdk_version: "0.1.0"
)
enqueue_all(client)

puts client.preview_json
response = client.shutdown(LogBrew::RecordingTransport.always_accept)

retry_client = LogBrew::Client.create(
  api_key: "LOGBREW_API_KEY",
  sdk_name: "logbrew-ruby",
  sdk_version: "0.1.0"
)
enqueue_all(retry_client)
retry_response = retry_client.flush(
  LogBrew::RecordingTransport.new([LogBrew::TransportError.network("temporary outage"), 202])
)

support_draft = LogBrew::SupportTicketDraft.create(
  source: "sdk",
  category: "ingest_failure",
  title: "Telemetry flush failed",
  description: "Flush returned usage_limit_exceeded",
  trace_id: "4BF92F3577B34DA6A3CE929D0E0E4736",
  diagnostics: {
    apiKey: "lbw_ingest_hidden",
    endpoint: "https://api.example/ingest?debug=true#frag",
    error: RuntimeError.new("contains hidden token")
  }
)

rejected_after_shutdown = false
begin
  client.action("evt_action_002", "2026-06-02T10:00:06Z", name: "deploy", status: "success")
rescue LogBrew::SdkError => error
  rejected_after_shutdown = error.code == "shutdown_error"
end

$stderr.puts JSON.generate(
  ok: rejected_after_shutdown,
  status: response.status_code,
  attempts: response.attempts,
  retryAttempts: retry_response.attempts,
  supportDraftRedacted: support_draft.dig("diagnostics", "apiKey") == "[redacted]",
  supportDraftTrace: support_draft.fetch("trace_id"),
  events: 6
)

File: java/logbrew-java/examples/RealUserSmoke.java
import co.logbrew.sdk.LogBrewClient;
import co.logbrew.sdk.RecordingTransport;
import co.logbrew.sdk.SdkException;
import co.logbrew.sdk.SupportTicketDraft;
import co.logbrew.sdk.TransportException;
import co.logbrew.sdk.TransportResponse;
import java.util.Map;

public final class RealUserSmoke {
    private RealUserSmoke() {
    }

    public static void main(String[] args) {
        LogBrewClient client = LogBrewClient.create("LOGBREW_API_KEY", "logbrew-java", "0.1.0");
        ReadmeExample.enqueueAll(client);

        System.out.println(client.previewJson());
        TransportResponse response = client.shutdown(RecordingTransport.alwaysAccept());

        LogBrewClient retryClient = LogBrewClient.create("LOGBREW_API_KEY", "logbrew-java", "0.1.0");
        ReadmeExample.enqueueAll(retryClient);
        TransportResponse retryResponse = retryClient.flush(RecordingTransport.scripted(
            TransportException.network("temporary outage"),
            Integer.valueOf(202)
        ));
        SupportTicketDraft supportDraft = SupportTicketDraft.create(SupportTicketDraft.Input
            .create("sdk", "ingest_failure", "Telemetry flush failed", "Flush returned usage_limit_exceeded")
            .runtime("java 21")
            .sdkPackage("co.logbrew:logbrew-sdk")
            .sdkVersion("0.1.0")
            .traceId("4BF92F3577B34DA6A3CE929D0E0E4736")
            .diagnostics(Map.of(
                "apiKey", "lbw_ingest_hidden",
                "endpoint", "https://api.example/ingest?debug=true"
            )));
        String supportDraftJson = supportDraft.toJson();
        boolean supportDraftRedacted = supportDraftJson.contains("[redacted]")
            && supportDraftJson.contains("[redacted-url]/ingest")
            && !supportDraftJson.contains("hidden")
            && !supportDraftJson.contains("api.example")
            && supportDraft.traceId().equals("4bf92f3577b34da6a3ce929d0e0e4736");

        boolean rejectedAfterShutdown = false;
        try {
            client.action(
                "evt_action_002",
                "2026-06-02T10:00:06Z",
                co.logbrew.sdk.ActionAttributes.create("deploy", "success")
            );
        } catch (SdkException error) {
            rejectedAfterShutdown = "shutdown_error".equals(error.code());
        }

        System.err.println(
            "{\"ok\":"
                + rejectedAfterShutdown
                + ",\"status\":"
                + response.statusCode()
                + ",\"attempts\":"
                + response.attempts()
                + ",\"retryAttempts\":"
                + retryResponse.attempts()
                + ",\"supportDraftRedacted\":"
                + supportDraftRedacted
                + ",\"events\":6}"
        );
    }
}
